### PR TITLE
Simplify type annotations in `visualize/_rank.py` and `visualization_tests/`

### PR DESCRIPTION
**Pull request by Alnusjaponica**
## Motivation
Follow up https://github.com/optuna/optuna/pull/4525, https://github.com/optuna/optuna/pull/4526, and https://github.com/optuna/optuna/pull/4427 for https://github.com/optuna/optuna/issues/4508.
Since `visualize/_rank.py` is added after the type annotation simplification of `visualize/` and `visualization_tests/`, neither the file nor its test is on the TODO list in the issue.

## Description of the changes
Simplify type annotation as follows:
> UP007: Fixes Union[x, y] to x | y
> UP006: Fixes collection types (e.g. List[str] -> list[str])
> F401: Removes unused imports such as from typing import List
> I002: Enforces from __future__ import annotations

### CODE REVIEW

**Comment by toshihikoyanase** [comment]
The PR seems to be review-ready. Do you have any TODO tasks?

**Comment by Alnusjaponica** [comment]
I noticed that type annotation simplification of `visualize_tests/` was reset as https://github.com/optuna/optuna/pull/4526 was removed from the history. Should I also apply simplification to all files in `visualize_tests/` or is it better to recover the changes separately?

**Comment by Alnusjaponica** [comment]
@toshihikoyanase 
I have decided that I re-apply the type annotation simplification in `visualization_tests/` too.

**Review by contramundum53 — APPROVED** [review]
LGTM.

**Comment by contramundum53** [comment]
@toshihikoyanase Could you review this PR?

File: optuna/visualization/_rank.py [hunk 1]
```diff
@@ -1,12 +1,10 @@
+from __future__ import annotations
+
 import math
 import typing
 from typing import Any
 from typing import Callable
-from typing import List
```
Comment (toshihikoyanase) [inline optuna/visualization/_rank.py:5]: We need to update `_TickInfo` 's members to remove `List`.

```
class _TickInfo(NamedTuple):
    coloridxs: List[float]
    text: List[str]
```

Comment (Alnusjaponica) [reply optuna/visualization/_rank.py:5]: Thanks for your comment. I updated them.

Comment (toshihikoyanase) [reply optuna/visualization/_rank.py:5]: Thanks!

**Review by toshihikoyanase — APPROVED** [review]
Thank you for your update. LGTM!

**Pull request merged**